qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteDiagnostics.java

```diff
@@ -1090,4 +1131,47 @@ private ResolvedJavaTypeInfo validateIterable(Part part, Section ownerSection,
 		}
 		return resolvedJavaType;
 	}
-}
+
+	private void validateCaseIsPart(Section caseIsSection, Section whenSwitchSection,
+			ResolvedJavaTypeInfo switchWhenType, String projectUri, ResolvingJavaTypeContext resolvingJavaTypeContext,
+			List<Diagnostic> diagnostics) {
+		String expression = whenSwitchSection.getExpressionContent();
+		List<Parameter> parameters = caseIsSection.getParameters();
+		// Iterate through children and ensure the type of the #case or #is parts are
+		// the same as the parent section
+		for (int i = 0; i < parameters.size(); i++) {
+			Parameter parameter = parameters.get(i);
+			String parameterName = parameter.getName();
+			if (i == 0 && "in".equals(parameterName) && parameters.size() > 1) {

```

Comment:
Replace with 

`if (i == 0 && parameters.size() > 1) {`

Indeed, as we can have several parameters (in, eq, etc, see https://github.com/quarkusio/quarkus/blob/9366bc89ad69ffd4ebef003b31b0e7cff46d8689/independent-projects/qute/core/src/main/java/io/quarkus/qute/WhenSectionHelper.java#L174)  

we need to take care of and it foolows the same logic than the real Qute parser:

https://github.com/quarkusio/quarkus/blob/9366bc89ad69ffd4ebef003b31b0e7cff46d8689/independent-projects/qute/core/src/main/java/io/quarkus/qute/WhenSectionHelper.java#L300